test/services/offer.service.test.js

```diff
@@ -681,50 +642,91 @@ describe('Offer Service', () => {
+        expect(paymentDates[0].date).to.eql(new Date('2021-03-01T00:00:00.000Z'));
+        expect(paymentDates[1].date).to.eql(new Date('2021-03-31T00:00:00.000Z'));
+        expect(paymentDates[2].date).to.eql(new Date('2021-04-30T00:00:00.000Z'));
+        expect(paymentDates[3].date).to.eql(new Date('2021-05-30T00:00:00.000Z'));
+        expect(paymentDates[4].date).to.eql(new Date('2021-06-29T00:00:00.000Z'));
+        expect(paymentDates[5].date).to.eql(new Date('2021-07-29T00:00:00.000Z'));
       });
     });
 
     context('when totalAmountPayable is equal to initial payment', () => {
+      const offer = OfferFactory.build(
+        {
+          enquiryId: enquiry._id,
+          vendorId: vendor._id,
+          totalAmountPayable: 100000,
+          initialPayment: 100000,
+          periodicPayment: 10000,
+          paymentFrequency: 30,
+          handOverDate: new Date('2021-03-01'),
+        },
+        { generateId: true },
+      );
       beforeEach(async () => {
-        await createOffer(offer2);
+        await createOffer(offer);
       });
       it('returns a valid accepted offer', async () => {
-        const paymentDates = generatePaymentSchedules(offer2);
+        const paymentDates = generatePaymentSchedules(offer);
         expect(paymentDates.length).to.eql(1);
         expect(paymentDates[0].amount).to.eql(100000);
-        expect(paymentDates[0].date).to.have.string('2020-12-11');
+        expect(paymentDates[0].date).to.eql(new Date('2021-03-01T00:00:00.000Z'));
       });
     });
 
     context('when  totalAmountPayable is slightly bigger than initial payment', () => {
+      const offer = OfferFactory.build(
+        {
+          enquiryId: enquiry._id,
+          vendorId: vendor._id,
+          totalAmountPayable: 100000,
+          initialPayment: 75000,
+          periodicPayment: 10000,
+          paymentFrequency: 14,
+          handOverDate: new Date('2021-03-01'),
+        },
+        { generateId: true },
+      );
       beforeEach(async () => {
-        await createOffer(offer3);
+        await createOffer(offer);
       });
+
       it('returns a valid accepted offer', async () => {
-        const paymentDates = generatePaymentSchedules(offer3);
-        expect(paymentDates.length).to.eql(2);
-        expect(paymentDates[0].amount).to.eql(95000);
-        expect(paymentDates[1].amount).to.eql(5000);
-        expect(paymentDates[0].date).to.have.string('2020-12-11');
-        expect(paymentDates[1].date).to.have.string('2020-12-25');
+        const paymentDates = generatePaymentSchedules(offer);
+        expect(paymentDates.length).to.eql(4);
+        expect(paymentDates[0].amount).to.eql(75000);
+        expect(paymentDates[1].amount).to.eql(10000);
+        expect(paymentDates[2].amount).to.eql(10000);
+        expect(paymentDates[3].amount).to.eql(5000);
+        expect(paymentDates[0].date).to.eql(new Date('2021-03-01T00:00:00.000Z'));
```

Comment:
We are already parsing the Date, so it should be safe to pass in a human readable date. Let's update this in all instance 🙏🏽 

```suggestion
        expect(paymentDates[0].date).to.eql(new Date('2021-03-01'));
```